src/satosa/frontends/openid_connect.py

```diff
@@ -173,6 +173,19 @@ def register_endpoints(self, backend_names):
         :rtype: list[(str, ((satosa.context.Context, Any) -> satosa.response.Response, Any))]
         :raise ValueError: if more than one backend is configured
         """
+        # See https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig
+        #
+        # We skip the scheme + host + port of the issuer URL, because we can only map the
+        # path for the provider config endpoint. We are safe to use urlparse().path here,
+        # because for issuer OIDC allows only https URLs without query and fragment parts.
+        issuer = self.provider.configuration_information["issuer"]
+        autoconf_path = ".well-known/openid-configuration"
+        provider_config = (
+            "^{}$".format(join_paths(urlparse(issuer).path.lstrip("/"), autoconf_path)),
```

Comment:
I vaguely remember that I wanted to differentiate between the configuration URL (which must start with the Issuer) and other endpoints (which in theory can be on different URLs), so that BASE=https://url1/path1 and configuration_information["issuer"]=https://url2/path2 would work.
But now I think this is just causing confusion, and I guess a weird setup like this would break anyway and should not be encouraged.

Reply:
Well, giving it some more thinking, I seem to remember why I did this originally.

Both SAML2 frontend and backend supports setting the entity_id to be entirely different from BASE. If I understand it right, also the OpenID backend supports this. I think they do it right, because in a basic flow, the entityID and Issuer are just two identifiers. They get an additional semantic meaning when metadata/provider discovery is in use: then they must be URLs, which point to a downloadable web resource, and hence SATOSA must provide the response.

I think, the current change is more consistent with the rest of the code than it would be if we hard coded BASE there.
What we could do is to support the '<base_url>' string replacement, what I don't really like, but... oh wait, isn't this _already_ supported? (See src/satosa/plugin_loader.py:235. It's surprising, to say at least.)